scripts/_download.js

```diff
@@ -203,3 +203,169 @@ export async function downloadAndExtractTarGz({ targetDirectory, url, sha256 })
+    throw error;
+  }
+}
+
+/**
+ * Extract an artifact ZIP file.
+ *
+ * @param {string} filePath - Path to the ZIP file
+ * @param {string} tempDir - Temporary directory for extraction
+ * @param {string} finalDir - Final directory for extracted files
+ */
+export async function extractArtifact(filePath, tempDir, finalDir) {
+  try {
+    console.log(`Checking if ${filePath} is a valid ZIP archive.`);
+
+    // First extraction to temporary directory
+    console.log(`Extracting ${filePath} to temporary directory: ${tempDir}`);
+    await unzipper.Open.file(filePath).then((d) => d.extract({ path: tempDir }));
+
+    // Find and extract any nested zip files in the temp directory
+    const files = await fs.readdir(tempDir);
+    for (const file of files) {
+      if (file.endsWith(".zip")) {
+        const nestedZipPath = join(tempDir, file);
+        console.log(`Extracting nested zip: ${nestedZipPath} to ${finalDir}`);
+        await unzipper.Open.file(nestedZipPath).then((d) => d.extract({ path: finalDir }));
+        console.log(`Deleting nested zip: ${nestedZipPath}`);
+        await fs.unlink(nestedZipPath);
+      }
+    }
+
+    // Cleanup temporary directory
+    console.log(`Cleaning up temporary directory: ${tempDir}`);
+    await fs.rm(tempDir, { recursive: true, force: true });
+
+    // Cleanup the original zip file
+    console.log(`Deleting original zip file: ${filePath}`);
+    await fs.unlink(filePath);
+
+    console.log(`Extraction complete: ${filePath} -> ${finalDir}`);
+  } catch (error) {
+    console.error(`Error extracting artifact ${filePath}:`, error.message);
+    throw error;
+  }
+}
+
+/**
+ * Download and process workflow artifacts.
+ *
+ * @param {string} targetDirectory - Directory to store downloaded artifacts
+ * @param {string} metaFile - Path to the metadata file
+ * @param {string} url - Base URL of the GitHub repository
+ * @param {string} workflow - Name of the workflow file
+ * @param {Array} assets - List of assets to downloa
+ */
+export async function downloadWorkflowArtifacts({
+  targetDirectory,
+  metaFile,
+  url,
+  workflow,
+  assets,
+}) {
+  const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
+
+  if (!GITHUB_TOKEN) {
+    console.error("Error: GITHUB_TOKEN environment variable is not set.");
+    process.exit(1);
+  }
+
+  try {
+    console.log("Fetching first successful workflow run...");
+    const workflowRunId = await fetchFirstSuccessfulRun(url, workflow, GITHUB_TOKEN);
+
+    if (!workflowRunId) {
+      throw new Error("No successful workflow runs found.");
+    }
+
+    console.log(`Workflow Run ID: ${workflowRunId}`);
+    const artifactUrls = await fetchArtifactsForRun(url, workflowRunId, GITHUB_TOKEN);
+
+    if (!artifactUrls || artifactUrls.length === 0) {
+      throw new Error("No artifacts found for the workflow run.");
+    }
+
+    const metadata = {
+      workflowRunId,
+      collectedAt: new Date().toISOString(),
+      assets: [],
+    };
```

Comment:
To match `downloadGitHubRelease()`, this metadata can look more like (assuming some more details are passed in as function params or returned from the fetches) :
```js
    const metadata = {
      org,
      repo,
      workflow,
      workflowRunId: workflowInfo.id,
      commitId: workflow.commitId,
      collectedAt: new Date().toISOString(),
      assets: [],
    };
```

Reply:
updated